### PR TITLE
Integrate the claims module

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

**Comment by gavofyork** [comment]
CC @jacogr

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -393,6 +403,7 @@ construct_runtime!(
 		Treasury: treasury::{Module, Call, Storage, Event<T>},
 		Parachains: parachains::{Module, Call, Storage, Config<T>, Inherent, Origin},
 		Slots: slots::{Module, Call, Storage, Event<T>},
+		Claims: claims::{Module, Call, Storage, Event<T>},
```
Comment (thiolliere) [inline runtime/src/lib.rs]: shouldn't it add ValidateUnsigned ?
```suggestion
		Claims: claims::{Module, Call, Storage, Event<T>, ValidateUnsigned},
```

**Comment by jacogr** [comment]
The metadata fights against this a bit atm, for this -

```rust
decl_event!(
	pub enum Event<T> where
		B = BalanceOf<T>,
		A = <T as system::Trait>::AccountId
	{
		/// Someone claimed some DOTs.
		Claimed(A, EthereumAddress, B),
	}
);
```

It returns types "A" and "B" (which really, could be anything) in the metadata. We don't get `A = AccountId`, raher only `A`. It would be better doing (based on the current limitations),

```rust
decl_event!(
	pub enum Event<T> where
		BalanceOf = BalanceOf<T>,
		AccountId = <T as system::Trait>::AccountId
	{
		/// Someone claimed some DOTs.
		Claimed(AccountId, EthereumAddress, BalanceOf),
	}
);
```

**Comment by gavofyork** [comment]
@jacogr make a PR?

**Comment by gavofyork** [comment]
nm already fixed.

**Pull request merged**